src/general/GameWorld.cs

```diff
@@ -26,6 +26,15 @@ public class GameWorld
     [JsonProperty]
     private Dictionary<uint, Species> worldSpecies = new Dictionary<uint, Species>();
 
+    [JsonProperty]
+    private Dictionary<double, List<WorldEventDescription>> worldTimeline = new();
+
+    /// <summary>
+    ///   Another place for worldTimeline keys to allow removing old dictionary keys
+    ///   through Queue's functionality
+    /// </summary>
+    private Queue<double> worldTimelineKeys = new();
```

Comment:
Does this need JSON saving? Also if the timeline uses just numbers to order events, can't we just sort the keys and take from that sorted list to find old items to be removed?

Reply:
After some further thought I think it makes sense to store this timeline events in individual patches. That way it will make it easier to support a patchwide-only filter.

Reply:
In the end having GameWorld store its own logs seems the best way and your sort keys and remove solution works great.